modules/video/src/ecc.cpp

```diff
@@ -397,13 +398,26 @@ double cv::findTransformECC(InputArray templateImage,
     Mat templateFloat = Mat(hs, ws, CV_32F);// to store the (smoothed) template
     Mat imageFloat    = Mat(hd, wd, CV_32F);// to store the (smoothed) input image
     Mat imageWarped   = Mat(hs, ws, CV_32F);// to store the warped zero-mean input image
-    Mat allOnes		= Mat::ones(hd, wd, CV_8U); //to use it for mask warping
     Mat imageMask		= Mat(hs, ws, CV_8U); //to store the final mask
 
+    Mat inputMaskMat = inputMask.getMat();
+    //to use it for mask warping
+    Mat preMask =
+      inputMaskMat.empty() ? Mat::ones(hd, wd, CV_8U) : inputMaskMat;
```

Comment:
you can not expect that the input mask is filled with 1's. OpenCV convention is that it only matters if certain mask pixel is equal to zero or not. Therefore, you should probably apply threshold() function and set non-zero pixels to 1 (but, of course, you should not touch the input mask, you should copy it to temporary array). In other words, you should probably write something like:

<pre>
Mat preMask;
if(inputMask.empty())
    preMask = Mat::ones(hd, wd, CV_8U);
else
   threshold(inputMask, preMask, 0, 1, THRESH_BINARY);
</pre>